src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/CSharpDocumentExcerptServiceTests.cs

```diff
@@ -0,0 +1,274 @@
+                    Assert.Equal("foo", result.Value.Content.GetSubText(c.TextSpan).ToString());
+                },
+                c =>
+                {
+                    Assert.Equal(ClassificationTypeNames.Text, c.ClassificationType);
+                    Assert.Equal(" ", result.Value.Content.GetSubText(c.TextSpan).ToString());
+                },
+                c =>
+                {
+                    Assert.Equal(ClassificationTypeNames.Operator, c.ClassificationType);
+                    Assert.Equal("=", result.Value.Content.GetSubText(c.TextSpan).ToString());
+                },
+                c =>
+                {
+                    Assert.Equal(ClassificationTypeNames.Text, c.ClassificationType);
+                    Assert.Equal(" ", result.Value.Content.GetSubText(c.TextSpan).ToString());
+                },
+                c =>
+                {
+                    Assert.Equal(ClassificationTypeNames.StringLiteral, c.ClassificationType);
+                    Assert.Equal("\"Hello, World!\"", result.Value.Content.GetSubText(c.TextSpan).ToString());
+                },
+                c =>
+                {
+                    Assert.Equal(ClassificationTypeNames.Punctuation, c.ClassificationType);
+                    Assert.Equal(";", result.Value.Content.GetSubText(c.TextSpan).ToString());
+                });
+        }
+
+        [Fact]
+        public async Task TryGetExcerptInternalAsync_SingleLine_CanClassifyCSharp_ImplicitExpression()
+        {
+            // Arrange
+            var razorSource = @"
+<html>
+@{
+    var foo = ""Hello, World!"";
+}
+  <body>@|foo|</body>
+  <div>@(3 + 4)</div><div>@(foo + foo)</div>
+</html>
+";
+
+            var (generatedDocument, razorSourceText, primarySpan, generatedSpan) = await InitializeAsync(razorSource);
+
+            var excerptService = new CSharpDocumentExcerptService();
+            var mappedLinePositionSpan = razorSourceText.Lines.GetLinePositionSpan(primarySpan);
+
+            // Act
+            var result = await excerptService.TryGetExcerptInternalAsync(
+                generatedDocument,
+                generatedSpan,
+                ExcerptModeInternal.SingleLine,
+                razorSourceText,
+                mappedLinePositionSpan,
+                CancellationToken.None);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(generatedSpan, result.Value.Span);
+            Assert.Same(generatedDocument, result.Value.Document);
+
+            // Verifies that the right part of the primary document will be highlighted.
+            Assert.Equal(
+                (await generatedDocument.GetTextAsync()).GetSubText(generatedSpan).ToString(),
+                result.Value.Content.GetSubText(result.Value.MappedSpan).ToString(),
+                ignoreLineEndingDifferences: true);
+
+            Assert.Equal(@"  <body>@foo</body>", result.Value.Content.ToString(), ignoreLineEndingDifferences: true);
```

Comment:
Not doing ClassifiedSpans validation for the rest as this is an area likely to change in the future once we no longer classify HTML through the C# classifier.